src/pages/views/PageDetails.tsx

```diff
@@ -59,22 +74,99 @@ export const PageDetails: React.FC<PageDetailsProps> = ({ id, params }) => {
     }
   };
 
+  const [
+    deleteAttributeValue,
+    deleteAttributeValueOpts
+  ] = useAttributeValueDeleteMutation({});
+
   return (
     <TypedPageRemove variables={{ id }} onCompleted={handlePageRemove}>
       {(pageRemove, pageRemoveOpts) => (
         <TypedPageUpdate>
           {(pageUpdate, pageUpdateOpts) => (
             <TypedPageDetailsQuery variables={{ id }}>
               {pageDetails => {
-                const handleUpdate = async (data: PageData) => {
+                const handleUpdate = async (data: PageSubmitData) => {
+                  let errors: Array<
+                    | AttributeErrorFragment
+                    | UploadErrorFragment
+                    | PageErrorFragment
+                  > = [];
+
+                  const uploadFilesResult = await Promise.all(
+                    data.attributesWithNewFileValue.map(fileAttribute =>
+                      uploadFile({
+                        variables: {
+                          file: fileAttribute.value
+                        }
+                      })
+                    )
+                  );
+
+                  const attributesWithAddedNewFiles: AttributeValueInput[] = uploadFilesResult.reduce(
+                    (attributesWithAddedFiles, uploadFileResult, index) => {
+                      const attribute = data.attributesWithNewFileValue[index];
+
+                      errors = [
+                        ...errors,
+                        ...uploadFileResult.data.fileUpload.uploadErrors
+                      ];
+                      return [
+                        ...attributesWithAddedFiles,
+                        {
+                          file:
+                            uploadFileResult.data.fileUpload.uploadedFile.url,
+                          id: attribute.id,
+                          values: []
+                        }
+                      ];
+                    },
+                    []
+                  );
+
                   const result = await pageUpdate({
                     variables: {
                       id,
-                      input: createPageInput(data)
+                      input: createPageInput(data, attributesWithAddedNewFiles)
                     }
                   });
 
-                  return result.data.pageUpdate.errors;
+                  errors = [...errors, ...result.data.pageUpdate.errors];
+
+                  if (errors.length === 0) {
+                    const deleteAttributeValuesResult = await Promise.all(
+                      pageDetails?.data?.page?.attributes.map(
+                        existingAttribute => {
+                          const fileValueUnused =
```

Comment:
would rather have this as a function in utils so it's not adding to the lots of logic that's already here